addons/website_sale_loyalty/controllers/portal.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http, _
+from odoo.http import request
+from odoo.addons.portal.controllers import portal
+from odoo.addons.portal.controllers.portal import pager as portal_pager
+
+class CustomerPortal(portal.CustomerPortal):
+
+    def _prepare_home_portal_values(self, counters):
+        values = super()._prepare_home_portal_values(counters)
+        partner = request.env.user.partner_id
+
+        SaleOrder = request.env['sale.order']
+        if 'loyalty_count' in counters:
+            values['loyalty_count'] = SaleOrder.search_count([
+                ('message_partner_ids', 'child_of', [partner.commercial_partner_id.id]),
+                ('state', 'in', ['sent', 'sale', 'done']),
+                ('website_id', '=', request.website.id),
+                '|', ('won_loyalty_points', '>', 0), ('spent_loyalty_points', '>', 0),
+            ]) if SaleOrder.check_access_rights('read', raise_exception=False) else 0
```

Comment:
Yes but `search_count` raises an `AccessError` if the right is not available.
This approach is reproduced from other implementations of `_prepare_home_portal_values(...)`.
This appears to be done to avoid a failure when accessing the home portal with users having different rights that do not give access to specific types.